sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -105,6 +107,15 @@ private[sql] class JSONRelation(
     jsonSchema
   }
 
+  override private[sql] def buildScan(
+      requiredColumns: Array[String],
+      filters: Array[Filter],
+      inputPaths: Array[String],
+      broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
As the files under the table path probably changed by other applications, I think we have to refresh the file status before the file scanning.

Probably we can move out the parquet meta data refreshing from the `HadoopFsRelation.refresh`.